equinox/_module.py

```diff
@@ -478,11 +481,69 @@ def __new__(
                                     "attempting to override "
                                     f"`{base.__module__}.{base.__qualname__}.{k}`."
                                 )
+
         # [Step 6] Register as a pytree.
+        data_fs = tuple(
+            f.name
+            for f in dataclasses.fields(cls)
+            if not f.metadata.get("static", False)
+        )
+        static_fs = tuple(
+            f.name for f in dataclasses.fields(cls) if f.metadata.get("static", False)
+        )
+
+        sntl = _flatten_sentinel
+
+        def flatten_module(
+            obj: "Module", /
+        ) -> tuple[tuple[PyTree, ...], _FlattenedData]:
+            """Flattens a module into a flattened representation."""
+            d = obj.__dict__
+
+            # Build dynamic keys, values in single pass
+            dynamic_fs = []
+            dynamic_vs = []
+            for k in data_fs:
+                v = d.get(k, sntl)
+                if v is sntl:
+                    continue
+                dynamic_fs.append(k)
+                dynamic_vs.append(v)
+
+            aux = _FlattenedData(
+                tuple(dynamic_fs),
+                tuple([(k, d.get(k, sntl)) for k in static_fs]),
+                tuple([(k, d.get(k, sntl)) for k in _wrapper_field_names]),
```

Comment:
Nit: this is materialising a list and then converting it to a tuple. I think a direct `tuple((k, d.get(k, sntl) for k in static_fs)` should presumably save a small amount.

Reply:
One would think. But I've timed it out for different stuff and haven't found that to be the case. Here's a quick test:

![CleanShot 2025-04-13 at 17 44 27@2x](https://github.com/user-attachments/assets/0cad0000-bcbb-4549-967e-e0d8296914ed)

Reply:
Mysterious! Well fair enough then :)